cli/src/commands/tui/ui/welcome.rs

```diff
@@ -0,0 +1,174 @@
+                [
+                    Constraint::Percentage(30),
+                    Constraint::Percentage(40),
+                    Constraint::Percentage(30),
+                ]
+                .as_ref(),
+            )
+            .split(center.inner(&Margin {
+                vertical: 1,
+                horizontal: 1,
+            }))
+    }
+
+    fn draw_main_block<B: Backend>(frame: &mut Frame<B>, center: Rect) {
+        let main_block = WidgetBlock::default()
+            .title(" Wukong CLI ")
+            .title_alignment(Alignment::Center)
+            .borders(Borders::ALL)
+            .padding(Padding::uniform(1));
+
+        frame.render_widget(main_block, center);
+    }
+
+    fn draw_version_info<B: Backend>(frame: &mut Frame<B>, top: Rect) {
+        let version_info_block = WidgetBlock::default()
+            .title(format!("Version: {}", env!("CARGO_PKG_VERSION")))
+            .title_alignment(Alignment::Center)
+            .padding(Padding::new(1, 1, 1, 0))
+            .style(Style::default().fg(Color::DarkGray));
+
+        frame.render_widget(version_info_block, top);
+    }
+
+    fn draw_verification_info<B: Backend>(app: &App, frame: &mut Frame<B>, layout: Rect) {
+        let okta_status =
+            Self::get_verification_status(app.state.is_okta_authenticated, "Okta token");
+        let gcloud_status =
+            Self::get_verification_status(app.state.is_gcloud_authenticated, "GCloud token");
+
+        let application_widget = Paragraph::new(vec![
+            Line::from(Self::format_status_line(okta_status)),
+            Line::from(Self::format_status_line(gcloud_status)),
+        ])
+        .alignment(Alignment::Center)
+        .wrap(Wrap { trim: true });
+
+        frame.render_widget(application_widget, layout);
+    }
+
+    fn draw_redirect_block<B: Backend>(app: &App, frame: &mut Frame<B>, layout: Rect) {
+        if let Some(timer) = app.state.welcome_screen_timer {
+            let time_remaining = timer.saturating_duration_since(Instant::now());
+
+            let redirect_block = Paragraph::new(vec![Line::styled(
+                format!("Redirecting in {} seconds...", time_remaining.as_secs()),
+                Style::default().fg(Color::DarkGray),
+            )])
+            .alignment(Alignment::Center)
+            .wrap(Wrap { trim: true });
+
+            frame.render_widget(redirect_block, layout);
+        }
+    }
+
+    fn draw_auth_instructions<B: Backend>(app: &App, frame: &mut Frame<B>, layout: Rect) {
+        if let (Some(gcloud_authenticated), Some(okta_authenticated)) = (
+            app.state.is_gcloud_authenticated,
+            app.state.is_okta_authenticated,
+        ) {
+            let gcloud_auth_block = Paragraph::new(vec![
+                Line::styled(
+                    Self::get_auth_instruction(
+                        okta_authenticated,
+                        "Please run `wukong init` to authenticate again. Press `q` to exit.",
```

Comment:
I think this message is a little bit misleading. It will always ask user to re-init even if they are already done the initialisation step. You can see it in the screenshot below. 

![image](https://github.com/mindvalley/wukong-cli/assets/1546109/72e4b1f0-2d65-4c4c-9c53-338954fab372)